File: backend/src/main/java/com/autotest/entity/ResultRule.java
package com.autotest.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * 测试结果解析规则
 */
@Data
@TableName("result_rules")
public class ResultRule {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 关联脚本ID（NULL表示全局规则）
     */
    private Long scriptId;

    /**
     * 规则名称
     */
    private String name;

    /**
     * 描述
     */
    private String description;

    /**
     * 解析方式: builtin=内置规则, script=解析脚本
     */
    private String parserType;

    /**
     * 内置规则格式: key_value, json
     */
    private String builtinFormat;

    /**
     * 脚本来源: package=从脚本包选择, inline=直接编写
     */
    private String scriptSource;

    /**
     * 脚本包中的路径
     */
    private String scriptPath;

    /**
     * 内联脚本内容
     */
    @TableField(typeHandler = com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler.class)
    private String scriptContent;

    /**
     * 脚本语言: python, shell
     */
    private String scriptLanguage;

    /**
     * 输入来源: stdout=标准输出, file=指定文件
     */
    private String inputSource;

    /**
     * 文件路径正则表达式
     */
    private String filePattern;

    /**
     * 输出格式: json, csv
     */
    private String outputFormat;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 优先级（用于规则排序）
     */
    private Integer priority;

    /**
     * 规则配置（JSON格式，用于旧的判定规则）
     */
    @TableField(typeHandler = com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler.class)
    private Map<String, Object> rules;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/resultRule.ts
import request from '@/utils/request'

export interface ResultRule {
  id: number
  scriptId: number | null
  name: string
  description: string
  parserType: 'builtin' | 'script'
  builtinFormat: 'key_value' | 'json' | null
  scriptSource: 'package' | 'inline' | null
  scriptPath: string | null
  scriptContent: string | null
  scriptLanguage: 'python' | 'shell' | null
  inputSource: 'stdout' | 'file'
  filePattern: string | null
  outputFormat: 'json' | 'csv'
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export interface TestParseParams {
  parserType: 'builtin' | 'script'
  builtinFormat?: 'key_value' | 'json'
  scriptSource?: 'package' | 'inline'
  scriptContent?: string
  scriptLanguage?: 'python' | 'shell'
  sampleInput: string
}

export const resultRuleApi = {
  /**
   * 获取脚本的解析规则列表
   */
  listByScript(scriptId: number) {
    return request.get<ResultRule[]>(`/result-rules/script/${scriptId}`)
  },

  /**
   * 获取全局解析规则列表
   */
  listGlobal() {
    return request.get<ResultRule[]>('/result-rules/global')
  },

  /**
   * 获取解析规则详情
   */
  getById(id: number) {
    return request.get<ResultRule>(`/result-rules/${id}`)
  },

  /**
   * 创建解析规则
   */
  create(data: Partial<ResultRule>) {
    return request.post<ResultRule>('/result-rules', data)
  },

  /**
   * 更新解析规则
   */
  update(id: number, data: Partial<ResultRule>) {
    return request.put<ResultRule>(`/result-rules/${id}`, data)
  },

  /**
   * 删除解析规则
   */
  delete(id: number) {
    return request.delete(`/result-rules/${id}`)
  },

  /**
   * 启用/禁用规则
   */
  setEnabled(id: number, enabled: boolean) {
    return request.put(`/result-rules/${id}/enabled`, { enabled })
  },

  /**
   * 测试解析规则
   */
  testParse(params: TestParseParams) {
    return request.post<Record<string, any>>('/result-rules/test', params)
  },
}
